ingest_data.R

```diff
@@ -0,0 +1,238 @@
+set.schema <- dbSendStatement(con, "set search_path to 'schemacrime';")
+
+library(testthat)
+
+
+# For logging the error messages and warnings
+sink("All_Output.log", type=c("output", "message"), append=TRUE, split=TRUE)
+
+
+# Reading in the data from the shell
+# It is important that it uses the first column as the rownames and does not 
+# create an extra column. This is taken care of in Filter_DATA.R by not writing the rownames.
+blotter.input <- read.csv("stdin", header = TRUE)
+
+
+
+# Function to adjust the neighborhood names-----
+
+# First we need a function that can match misspelled neighborhoods to the real neighborhood
+# name used and stored in the data base:
+accepted.hoods <- dbGetQuery(con, "select * from schemaCrime.hoods;")$hood
+
+# I found a lot of Golden Triangle/Civic Arena, but it does not occur in the list:
+accepted.civic <- dbGetQuery(con, "SELECT * FROM hoods WHERE hood
+LIKE '%Civic%'") # There is really nothing with Civic Arena.
+accepted.golden <- dbGetQuery(con, "SELECT * FROM hoods WHERE hood
+LIKE '%Golden%'") # And nothing with golden
+# It will not add Golden Triangle/Civic Arena to the list, it will be filtered out as invalid.
+
+# We need to make the data frame neighborhoods not factors but just strings
+blotter.input$NEIGHBORHOOD <- as.character(blotter.input$NEIGHBORHOOD)
+
+
+
+changed.count <- 0 #Initializing global counter for number of rows changed
+# This function will look for misspellings or fragments of neighborhood names and replace them
+# with the correct one. It will also count how many times it changed a name.
+# Input: neighborhood name
+# Output: corrected neighborhood name or intial input if there is no similar name available
+#         and the name should be invalid
+replace.neighborhood.name <- function(input.name, input.rowid){
+    if(input.name %in% accepted.hoods){warning("The neighborhood name should have been accepted.")}
+    if(!(input.name %in% accepted.hoods) &&  
+       input.name!="Golden Triangle/Civic Arena"){ 
+      # we don't need to see Golden Triangle here
+      
+      print(input.name)
+      
+      if(tolower(input.name) %in% tolower(accepted.hoods)){
+        replacement.hoods <- accepted.hoods[which(tolower(accepted.hoods)==tolower(input.name))]
+      }else{
+        # We replace it with the first one of the similar neighborhoods, 
+        # but print a warning if there are several
+        replacement.hoods <- accepted.hoods[grepl(substr(tolower(input.name), 
+                                                         start = 4, stop = 7), tolower(accepted.hoods))]
+        }
+      
+     
+      
+      if(length(replacement.hoods)>1){
+        print(paste("For ID", input.rowid, "there are multiple matching neighborhoods, which are", replacement.hoods))
+      }
+      print(paste("replaced with", replacement.hoods[1]))}
+  
+  if(input.name=="Golden Triangle/Civic Arena"){
+    return(input.name)
+    }else if(is.na(replacement.hoods[1])){
```

Comment:
Indentation is off at quite a few places in this function. Best to place the closing brace of an if-statement at the level of `if`.